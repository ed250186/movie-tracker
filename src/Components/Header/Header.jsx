import React from "react";
import { BrowserRouter as Router, Route, NavLink, Link } from "react-router-dom";
import "./Header.scss";
import SignIn from "../SignIn/SignIn.jsx";
import SignUp from "../SignUp/SignUp.jsx";

const Header = () => {
  return (
    <header>
      <Router>
        <Link to="/">
          <h1>Movie Tracker</h1>
        </Link>
      </Router>
      <Router>
        <div className="btn-container">
          <NavLink to="/signin" className="style-header-btns">
            <button>SignIn</button>
          </NavLink>
          <NavLink to="/signup" className="style-header-btns">
            <button>SignUp</button>
          </NavLink>
        </div>
        <Route exact path="/signin" component={SignIn} />
        <Route exact path="/signup" component={SignUp} />
      </Router>
    </header>
  );
};

export default Header;
