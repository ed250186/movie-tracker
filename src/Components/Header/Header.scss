header {
  height: 10%;
  width: 100%;
  button {
    color: blue;
  }
}