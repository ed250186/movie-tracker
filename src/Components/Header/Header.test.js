import React from 'react';
import {shallow, mount} from 'enzyme';
import {Header} from './Header';
import {BrowserRouter as Router} from 'react-router-dom'

// 81, 100, 62, 91

// line 58

describe('Header', () => {
  let wrapper, props
  let event = {
    preventDefault: jest.fn()
  }

  beforeEach(() => {
    props = {
      login: {
        loggedIn: true,
        name: 'Nathan'
      },
      signOutUser: jest.fn(),

    }
    wrapper = shallow(
        <Header {...props}/>
    )
  })

  it('should match snapshot logged out', () => {
    wrapper.setProps({
      login: {
        loggedIn: false,
        name: 'Nathan'
      }
    })
    expect(wrapper).toMatchSnapshot()
  })

  it('should match snapshot logged in', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should be called with an event', () => {
    wrapper.instance().signOut = jest.fn()
    wrapper.find('.signOut').simulate('click')
    expect(wrapper.instance().signOut).toHaveBeenCalled()
  })

  it('should be called with an event', () => {
    wrapper.instance().signOut = jest.fn()
    wrapper.find('.signOut').simulate('click')
    expect(wrapper.instance().signOut).toHaveBeenCalled()
  })

  it('should call signOutUser', () => {
    wrapper.instance().signOut(event);
    expect(props.signOutUser).toHaveBeenCalled()
  })

})