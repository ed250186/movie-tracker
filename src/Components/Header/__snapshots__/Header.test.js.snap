// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot logged in 1`] = `
<header>
  <NavLink
    to="/"
  >
    <h1>
      Movie Tracker
    </h1>
  </NavLink>
  <div
    className="btn-container"
  >
    <p
      className="welcome"
    >
      Welcome back, Nathan!
    </p>
    <button
      className="signOut"
      onClick={[Function]}
    >
      SignOut
    </button>
  </div>
</header>
`;

exports[`Header should match snapshot logged out 1`] = `
<header>
  <NavLink
    to="/"
  >
    <h1>
      Movie Tracker
    </h1>
  </NavLink>
  <div
    className="btn-container"
  >
    <p
      className="welcome"
    />
    <div>
      <NavLink
        className="style-header-btns"
        to="/signin"
      >
        <button>
          SignIn
        </button>
      </NavLink>
      <NavLink
        className="style-header-btns"
        to="/signup"
      >
        <button>
          SignUp
        </button>
      </NavLink>
    </div>
  </div>
</header>
`;
